pkg/build/api/v1/types.go

```diff
@@ -171,6 +176,31 @@ type BuildSource struct {
 	SourceSecret *kapi.LocalObjectReference `json:"sourceSecret,omitempty" description:"supported auth methods are: ssh-privatekey"`
 }
 
+// ImageSource describes an image that is used as source for the build
+type ImageSource struct {
+	// From is a reference to an ImageStreamTag, ImageStreamImage, or DockerImage to
+	// copy source from.
+	From kapi.ObjectReference `json:"from" description:"reference to ImageStreamTag, ImageStreamImage, or DockerImage"`
+
+	// Paths is a list of source and destination paths to copy from the image.
+	Paths []ImageSourcePath `json:"paths" description:"paths to copy from image"`
+
+	// PullSecret is a reference to a secret to be used to pull the image from a registry
+	// If the image is pulled from the OpenShift registry, this field does not need to be set.
+	PullSecret *kapi.LocalObjectReference `json:"pullSecret,omitempty" description:"overrides the default pull secret for the source image"`
```

Comment:
Future proofing when we add push secret?  Don't buy it.  Obvious to the
user?  Maybe buy it.

On Thu, Dec 17, 2015 at 1:20 PM, Ben Parees <notifications@github.com>
wrote:

> In pkg/build/api/v1/types.go
> <https://github.com/openshift/origin/pull/6324#discussion_r47940697>:
>
> > @@ -171,6 +176,31 @@ type BuildSource struct {
> >  	SourceSecret *kapi.LocalObjectReference `json:"sourceSecret,omitempty" description:"supported auth methods are: ssh-privatekey"`
> >  }
> >
> > +// ImageSource describes an image that is used as source for the build
> > +type ImageSource struct {
> > +	// From is a reference to an ImageStreamTag, ImageStreamImage, or DockerImage to
> > +	// copy source from.
> > +	From kapi.ObjectReference `json:"from" description:"reference to ImageStreamTag, ImageStreamImage, or DockerImage"`
> > +
> > +	// Paths is a list of source and destination paths to copy from the image.
> > +	Paths []ImageSourcePath `json:"paths" description:"paths to copy from image"`
> > +
> > +	// PullSecret is a reference to a secret to be used to pull the image from a registry
> > +	// If the image is pulled from the OpenShift registry, this field does not need to be set.
> > +	PullSecret *kapi.LocalObjectReference `json:"pullSecret,omitempty" description:"overrides the default pull secret for the source image"`
>
> future proofing and make it obvious to the user what the secret is used
> for?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6324/files#r47940697>.
>